const catchError = require('../utils/catchError');
const { tag } = require('../models');

const getAll = catchError(async (req, res) => {
  const results = await tag.findAll();
  return res.json(results);
});

const create = catchError(async (req, res) => {
  const result = await tag.create(req.body);
  return res.status(201).json(result);
});

const getOne = catchError(async (req, res) => {
  const { id } = req.params;
  const result = await tag.findByPk(id);
  if (!result) return res.sendStatus(404);
  return res.json(result);
});

const remove = catchError(async (req, res) => {
  const { id } = req.params;
  await tag.destroy({ where: { id } });
  return res.sendStatus(204);
});

const update = catchError(async (req, res) => {
  const { id } = req.params;
  const result = await tag.update(req.body, { where: { id }, returning: true });
  if (result[0] === 0) return res.sendStatus(404);
  return res.json(result[1][0]);
});

module.exports = {
  getAll,
  create,
  getOne,
  remove,
  update,
};
